integration_tests/test_rough_align.py

```diff
@@ -0,0 +1,382 @@
+
+import os
+import pytest
+import logging
+import renderapi
+import json
+import glob
+import marshmallow as mm
+from test_data import (ROUGH_MONTAGE_TILESPECS_JSON,
+                       ROUGH_POINT_MATCH_COLLECTION,
+                       ROUGH_DS_TEST_TILESPECS_JSON,
+                       render_params,
+                       test_rough_parameters as solver_example)
+
+from rendermodules.module.render_module import RenderModuleException
+from rendermodules.materialize.render_downsample_sections import RenderSectionAtScale, create_tilespecs_without_mipmaps
+from rendermodules.dataimport.make_montage_scapes_stack import MakeMontageScapeSectionStack
+from rendermodules.rough_align.do_rough_alignment import SolveRoughAlignmentModule
+from rendermodules.rough_align.apply_rough_alignment_to_montages import (ApplyRoughAlignmentTransform, 
+                                                                         example as ex1,
+                                                                         apply_rough_alignment)
+
+
+logger = renderapi.client.logger
+logger.setLevel(logging.DEBUG)
+
+
+@pytest.fixture(scope='module')
+def render():
+    render_params['project'] = 'rough_align_test'
+    render = renderapi.connect(**render_params)
+    return render
+
+
+@pytest.fixture(scope='module')
+def tspecs_from_json():
+    tilespecs = [renderapi.tilespec.TileSpec(json=d)
+        for d in ROUGH_MONTAGE_TILESPECS_JSON]
+    return tilespecs
+
+@pytest.fixture(scope='module')
+def tspecs():
+    tilespecs = [renderapi.tilespec.TileSpec(json=d)
+                for d in ROUGH_DS_TEST_TILESPECS_JSON]
+    return tilespecs
+
+# A stack with multiple sections montaged
+@pytest.fixture(scope='module')
+def montage_stack(render,tspecs_from_json):
+    test_montage_stack = 'input_montage_stack'
+    renderapi.stack.create_stack(test_montage_stack, render=render)
+    renderapi.client.import_tilespecs(test_montage_stack, 
+                                      tspecs_from_json, 
+                                      render=render)
+    renderapi.stack.set_stack_state(test_montage_stack, 
+                                    'COMPLETE', 
+                                    render=render)
+
+    # assure stack is built correctly
+    assert len({tspec.tileId for tspec
+                in tspecs_from_json}.symmetric_difference(set(
+                    renderapi.stack.get_stack_tileIds(
+                        test_montage_stack, render=render)))) == 0
+    
+    zvalues = render.run(renderapi.stack.get_z_values_for_stack,
+                          test_montage_stack)
+    zs = [1020, 1021, 1022]
+    assert(set(zvalues) == set(zs))
+
+    yield test_montage_stack
+    #renderapi.stack.delete_stack(test_montage_stack, render=render)
```

Comment:
i'm conflicted as to whether this should be uncommented.. on one hand its a nice idea to clean up after ourselves with tests and makes test easy to re-run without confusing results with the render server, on the other hand its annoying to debug what went wrong with the test when the stack is gone afterwards.  Interested to hear other people's thoughts about this... more than anything I think we should be consistent.

Reply:
This should not be commented... I am having it commented until I reach the code coverage.

Reply:
In this specific case... this comment is good as the stack is required till the end of the test and a reference once forwards the execution to the delete statement - not the right way I want the testing to be done.

Reply:
yield will cause the stack to stick around until all the tests that use this fixture are completed, so if you have a test that needs that stack to exist, simply referencing this feature will force it to stick around.  I wasn't sure if 'reference once forwards' means that you thought that the first time it was referenced it would proceed to delete?